.github/workflows/update_deps.yaml

```diff
@@ -0,0 +1,36 @@
+name: Update dependencies
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 12 * * MON"
+
+jobs:
+  update_deps:
+    name: Update dependencies
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python 3
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+
+      - name: Update dependencies
+        run: make deps/update
+
+      - name: Run pytest
+        run: make test
+
+      - name: Open pull request
+        uses: peter-evans/create-pull-request@v6
```

Comment:
Why not use Dependabot / Renovate for this? 
If I understand correctly, this action will commit everything that was changed in the previous steps. Who knows what is there? Dependabot or Renovate update one dependency at a time. In case of breaking changes, this would be easier to roll back, wouldn't it?

Let me know if you like the idea, I could help you to set this up.

Reply:
Yes, you're right. That's probably the proper way to go.